gems/aws-sdk-core/lib/aws-sdk-core/rest/request/endpoint.rb

```diff
@@ -30,30 +30,47 @@ def uri(base_uri, params)
         private
 
         def apply_path_params(uri, params)
-          path = uri.path.sub(/\/$/, '') + @path_pattern.split('?')[0]
+          path = uri.path.sub(%r{/$}, '') + @path_pattern.split('?')[0]
           uri.path = path.gsub(/{.+?}/) do |placeholder|
             param_value_for_placeholder(placeholder, params)
           end
         end
 
         def param_value_for_placeholder(placeholder, params)
           name = param_name(placeholder)
-          value = params[name].to_s
+          param_shape = @rules.shape.member(name).shape
+          value = case param_shape.is_a?(Seahorse::Model::Shapes::TimestampShape)
```

Comment:
Your case could be `case param_shape` and your when statement can be TimestampShape.